client/settings/action-creators.ts

```diff
@@ -1,56 +1,68 @@
 import { TypedIpcRenderer } from '../../common/ipc'
-import { LocalSettingsData, ScrSettingsData } from '../../common/local-settings'
-import { LOCAL_SETTINGS_SET_BEGIN, SCR_SETTINGS_SET_BEGIN } from '../actions'
+import { LocalSettings, ScrSettings } from '../../common/settings/local-settings'
 import { audioManager } from '../audio/audio-manager'
-import { openDialog } from '../dialogs/action-creators'
-import { DialogType } from '../dialogs/dialog-type'
 import { ThunkAction } from '../dispatch-registry'
-import { isStarcraftHealthy } from '../starcraft/is-starcraft-healthy'
+import { JsonLocalStorageValue } from '../local-storage'
+import { abortableThunk, RequestHandlingSpec } from '../network/abortable-thunk'
+import { openSnackbar } from '../snackbars/action-creators'
+import { ChangeSettingsSubPage, CloseSettings, OpenSettings } from './actions'
+import { AppSettingsSubPage, SettingsSubPage } from './settings-sub-page'
 
 const ipcRenderer = new TypedIpcRenderer()
 
-export function openSettingsDialog(): ThunkAction {
-  return (dispatch, getState) => {
-    const { starcraft } = getState()
-
-    if (!isStarcraftHealthy({ starcraft })) {
-      dispatch(
-        openDialog({
-          type: DialogType.StarcraftPath,
-        }),
-      )
-    } else {
-      dispatch(
-        openDialog({
-          type: DialogType.Settings,
-        }),
-      )
-    }
+// TODO(2Pac): Clear or migrate the previously saved tab in local storage to this?
+const savedSettingsSubPage = new JsonLocalStorageValue<SettingsSubPage>('settingsSubPage')
+
+export function openSettings(subPage?: SettingsSubPage): OpenSettings {
+  return {
+    type: '@settings/openSettings',
+    payload: {
+      subPage: subPage ?? savedSettingsSubPage.getValue() ?? AppSettingsSubPage.Sound,
```

Comment:
> Why sound?

Because it is rendered in the first position in the new settings navigation :d

I left a TODO to replace it with user account details page once I move that over to the new settings.